### DISCUSSION TITLE
manually Release glide image from memory

### DISCUSSION THREAD

**Discussion by renzified** [skydoves/landscapist#133] [Q&A]
How do we release the loaded image from the memory when using glide with jetpack compose?

**Comment by faraz152** [comment]
Glide manages memory automatically via its `MemoryCache`, so in most cases you should let it handle eviction. That said, there are a few ways to explicitly release images depending on what you need:

---

**1. Release the entire memory cache (e.g. on low memory)**

Must be called on the **main thread**:

```kotlin
// In your Activity/ViewModel responding to onTrimMemory:
Glide.get(context).clearMemory()
```

Glide already hooks into `ComponentCallbacks2` if you register it in your `Application`, so for low-memory situations you typically get this for free.

---

**2. Prevent a specific URL from being cached at all**

Use this in `GlideImage` when you know the image should never be held in memory:

```kotlin
GlideImage(
    imageModel = { url },
    imageOptions = ImageOptions(
        requestBuilder = {
            skipMemoryCache(true)
            diskCacheStrategy(DiskCacheStrategy.NONE)
        }
    )
)
```

---

**3. Invalidate a specific URL already in disk cache**

```kotlin
// Background thread
CoroutineScope(Dispatchers.IO).launch {
    Glide.get(context).clearDiskCache()
}

// Or invalidate a single key by signature:
val requestOptions = RequestOptions().signature(ObjectKey(System.currentTimeMillis()))
```

---

**4. The landscapist composable side**

When a `GlideImage` composable leaves the composition, landscapist automatically calls `Glide.with(context).clear(target)` via its internal `DisposableEffect` — so there is no memory leak for composables that are removed from the tree. You do not need to do anything special for the normal case.
